src/installer/tests/HostActivation.Tests/MultiArchInstallLocation.cs

```diff
@@ -0,0 +1,188 @@
+                });
+                Command.Create(appExe)
+                    .EnableTracingAndCaptureOutputs()
+                    .ApplyRegisteredInstallLocationOverride(registeredInstallLocationOverride)
+                    .DotNetRoot(null)
+                    .Execute()
+                    .Should().HaveStdErrContaining($"Found install location path '{path1}'.")
+                    .And.HaveStdErrContaining($"Found architecture-specific install location path: '{path1}' ('{arch1}').")
+                    .And.HaveStdErrContaining($"Found architecture-specific install location path matching the current OS architecture ('{arch1}'): '{path1}'.")
+                    .And.HaveStdErrContaining($"Found architecture-specific install location path: '{path2}' ('{arch2}').")
+                    .And.HaveStdErrContaining($"Using global installation location [{path1}] as runtime location.");
+            }
+        }
+
+        [Fact]
+        [SkipOnPlatform(TestPlatforms.Windows, "This test targets the install_location config file which is only used on Linux and macOS.")]
+        public void InstallLocationFile_OnlyFirstLineMayNotSpecifyArchitecture()
+        {
+            var fixture = sharedTestState.StandaloneAppFixture
+                .Copy();
+
+            var appExe = fixture.TestProject.AppExe;
+            using (var registeredInstallLocationOverride = new RegisteredInstallLocationOverride(appExe))
+            {
+                registeredInstallLocationOverride.SetInstallLocation(new (string, string)[] {
+                    (string.Empty, "a/b/c"),
+                    (string.Empty, "x/y/z"),
+                });
+                Command.Create(appExe)
+                    .EnableTracingAndCaptureOutputs()
+                    .ApplyRegisteredInstallLocationOverride(registeredInstallLocationOverride)
+                    .DotNetRoot(null)
+                    .Execute()
+                    .Should().HaveStdErrContaining($"Found install location path 'a/b/c'.")
+                    .And.HaveStdErrContaining($"Only the first line in '{registeredInstallLocationOverride.PathValueOverride}' may not have an architecture prefix.")
+                    .And.HaveStdErrContaining($"Using install location 'a/b/c'.");
+            }
+        }
+
+        public class SharedTestState : IDisposable
+        {
+            public string BaseDirectory { get; }
+            public TestProjectFixture StandaloneAppFixture { get; }
+            public RepoDirectoriesProvider RepoDirectories { get; }
+            public string InstallLocation { get; }
+
+            public SharedTestState()
+            {
+                RepoDirectories = new RepoDirectoriesProvider();
+                var fixture = new TestProjectFixture("StandaloneApp", RepoDirectories);
+                fixture
+                    .EnsureRestoredForRid(fixture.CurrentRid)
+                    .PublishProject(runtime: fixture.CurrentRid, selfContained: false);
```

Comment:
I don't know if this is still the case or if it even makes sense anymore, but I think `StandaloneApp` was generally used for self-contained testing (since it sets RuntimeIdentifier, so build/publish defaults to self-contained) and `PortableApp` was generally used for framework-dependent.